refactor(core): remove webworker related checks on `assertDomNode`

JeanMeche opened the pull request:
Since the drop of the webworker platform the node can't be a `WebWorkerRenderNode`. 

## PR Type
What kind of change does this PR introduce?

- [x] Refactoring (no functional changes, no api changes)

pkozlowski-opensource reviewed (changes requested):
I would rather keep it since we are unwrapping DOM nodes from view-related data structures. Safe to remove webworker related checks though.

JeanMeche committed:
refactor(core): remove webworker related checks on `assertDomNode`

Since the drop of the webworker platform the node can't be a `WebWorkerRenderNode`.

JeanMeche commented:
> I would rather keep it since we are unwrapping DOM nodes from view-related data structures. Safe to remove webworker related checks though.

Makes sense, PR update.

alxhub commented:
This PR was merged into the repository by commit 73fcf9f20bbecb45664a979522bbd527f5e5c28f.
